File: app/controllers/workouts.py
from flask import *
import database as db
import json

workouts = Blueprint('workouts', __name__, template_folder='views')

@workouts.route('/workout/add', methods=['POST'])
def add_workout_route():
  """ Add a workout """
  cur = db.get_cursor()
  
  data = request.get_json()
  user_id = data['user_id']
  date = data['date']
  short_name = data['short_name']
  weight = data['weight']
  reps = data['reps']
  sets = data['sets']
  notes = data['notes']

  print data
  print user_id
  # Make sure thelift exists in the database... really can't happen
  cur.execute("SELECT COUNT(*) AS count FROM lifts WHERE short_name = %s", [short_name])
  if cur.fetchone()['count'] == 0:
    abort(425)
  cur.execute("""
    INSERT INTO workouts 
    (user_id, workout_date, short_name, weight, reps, sets, notes)
    VALUES (%s, %s, %s, %s, %s, %s, %s);
    """, [user_id, date, short_name, weight, reps, sets, notes])

  return "Added the workout!" 

@workouts.route('/workouts/<string:user_id>')
@workouts.route('/workouts/<string:user_id>/all')
def get_all_workouts_route(user_id):
  """ Gets all of a users workouts """
  print session
  cur = db.get_cursor()
  sql = """
    SELECT name, workout_date, name, weight, reps, sets
    FROM workouts, lifts 
    WHERE workouts.short_name = lifts.short_name AND
      user_id = %s 
    ORDER BY workout_date
    """ % user_id
  print sql
  cur.execute(sql)
  results = cur.fetchall()
  print results
  for res in results:
    res['workout_date'] = str(res['workout_date'])
  print results
  return Response(json.dumps(results), mimetype='application/json')

@workouts.route('/workouts/recent')
def get_recent_workouts_route():
  """ Gets all of a users workouts """
  cur = db.get_cursor()
  data = {}
  data['best_lift'] = 200
  data['recent_lift'] = 100
  return jsonify(data)

File: src/api/workouts.php
<?php

declare(strict_types = 1);
require_once __DIR__ . '/../../vendor/autoload.php';
require_once __DIR__ . '/../database.php';

// Match an integer as id on:
// /api/workouts/max/USERID
$this->respond('GET', '/max/[i:id]',
  function ($request, $response) {
    $db = new PG_DB();
    $query = "SELECT MAX(weight) AS max FROM workouts WHERE user_id = '$request->id'";
    $result = ['best_lift' => $db->query($query)->fetch()['max']];
    $response->json($result);
  });

// Match an integer as id on:
// /api/workouts/recent/USERID
$this->respond('GET', '/recent/[i:id]',
  function ($request, $response) {
    $db = new PG_DB();
    $query = "SELECT weight FROM workouts WHERE user_id = '$request->id' ORDER BY workout_date LIMIT 1";
    $result = ['recent_lift' => $db->query($query)->fetch()['weight']];
    $response->json($result);
  });

$this->respond('POST', '/add',
  function ($request, $response) {
    $db = new PG_DB();
    $insert_stmt = 'INSERT INTO workouts(workout_date, user_id, short_name, weight, reps, sets, notes) VALUES';
    $insert_stmt .= '(:date, :user_id, :short_name, :weight, :reps, :sets, :notes)';
    $prepared_stmt = $db->prepare($insert_stmt);
    $prepared_stmt->execute(['date' => $date, 'short_name' => $short_name,
      'reps' => $reps, 'sets' => $sets, 'user_id' => $user_id, 'weight' => $weight]);
    $response->json(['status' => 'good']);
  });

